src/DocumentFormat.OpenXml/SimpleTypes/OpenXmlComparableValueType.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace DocumentFormat.OpenXml
+{
+    /// <summary>
+    /// Represents a comparable and equatable value for simple value types (Int32, UInt32, Byte, struct, etc).
+    /// </summary>
+    /// <typeparam name="T">The type of the value.</typeparam>
+    public abstract class OpenXmlComparableValueType<T> : OpenXmlSimpleValue<T>,
```

Comment:
What if `OpenXmlComparableValueType<T> : OpenXmlComparableReferenceType<T?>`? This could probably consolidate a bit of code.

Reply:
Actually, that's probably a breaking change (it would change the inheritance chain of some of the current values)

Reply:
Yeah that's good for now. We can get the functionality in, and then mark it for clean up with an issue opened against the v3.0 milestone